src/qt/psbtoperationsdialog.cpp

```diff
@@ -0,0 +1,255 @@
+    CAmount totalAmount = 0;
+    for (const CTxOut& out : psbtx.tx->vout) {
+        CTxDestination address;
+        ExtractDestination(out.scriptPubKey, address);
+        totalAmount += out.nValue;
+        tx_description.append(tr(" * Sends %1 to %2")
+            .arg(BitcoinUnits::formatWithUnit(BitcoinUnits::BTC, out.nValue))
+            .arg(QString::fromStdString(EncodeDestination(address))));
+        tx_description.append("<br>");
+    }
+
+    PSBTAnalysis analysis = AnalyzePSBT(psbtx);
+    tx_description.append(" * ");
+    if (!*analysis.fee) {
+        // This happens if the transaction is missing input UTXO information.
+        tx_description.append(tr("Unable to calculate transaction fee or total transaction amount."));
+    } else {
+        tx_description.append(tr("Pays transaction fee: "));
+        tx_description.append(BitcoinUnits::formatWithUnit(BitcoinUnits::BTC, *analysis.fee));
+
+        // add total amount in all subdivision units
+        tx_description.append("<hr />");
+        QStringList alternativeUnits;
+        for (const BitcoinUnits::Unit u : BitcoinUnits::availableUnits())
+        {
+            if(u != m_client_model->getOptionsModel()->getDisplayUnit())
+                alternativeUnits.append(BitcoinUnits::formatHtmlWithUnit(u, totalAmount));
+        }
+        tx_description.append(QString("<b>%1</b>: <b>%2</b>").arg(tr("Total Amount"))
+            .arg(BitcoinUnits::formatHtmlWithUnit(m_client_model->getOptionsModel()->getDisplayUnit(), totalAmount)));
+        tx_description.append(QString("<br /><span style='font-size:10pt; font-weight:normal;'>(=%1)</span>")
+            .arg(alternativeUnits.join(" " + tr("or") + " ")));
+    }
+
+    size_t num_unsigned = CountPSBTUnsignedInputs(psbtx);
+    if (num_unsigned > 0) {
+        tx_description.append("<br><br>");
+        tx_description.append(tr("Transaction has %1 unsigned inputs.").arg(QString::number(num_unsigned)));
+    }
+
+    return tx_description.toStdString();
+}
+
+void PSBTOperationsDialog::showStatus(const QString &msg, StatusLevel level) {
+    m_ui->statusBar->setText(msg);
+    switch (level) {
+        case StatusLevel::INFO: {
+            m_ui->statusBar->setStyleSheet("QLabel { background-color : lightgreen }");
+            break;
+        }
+        case StatusLevel::WARNING: {
+            m_ui->statusBar->setStyleSheet("QLabel { background-color : orange }");
+            break;
+        }
+        case StatusLevel::ERROR: {
+            m_ui->statusBar->setStyleSheet("QLabel { background-color : red }");
+            break;
+        }
+    }
+    m_ui->statusBar->show();
+}
+
+void PSBTOperationsDialog::showTransactionStatus(const PartiallySignedTransaction &psbtx) {
+    PSBTAnalysis analysis = AnalyzePSBT(psbtx);
+
+    switch (analysis.next) {
+        case PSBTRole::UPDATER: {
+            showStatus(tr("Transaction is missing some information about inputs."), StatusLevel::WARNING);
```

Comment:
I wonder if these strings can be moved to a place where the RPC can also use them.

Reply:
That would make sense actually, yeah. I could move them to wherever AnalyzePSBT is. What would the impact of this be on translation issues? I know that tr() is only used for GUI strings, and _() is used for non-GUI strings -- is there precedent for shared strings?

Reply:
I'm not sure actually. @laanwj, @hebasto?

Reply:
@gwillen 
> I know that tr() is only used for GUI strings, and _() is used for non-GUI strings -- is there precedent for shared strings?

Actually, `bilingual_str` type from `util/translation.h` supports both original and translated strings. More details are available in #16362 and #16224.

You could use something like this:

```
#include <util/translation.h>

bilingual_str bs = _("Transaction is missing some information about inputs.");
showStatus(bs.translated, StatusLevel::WARNING);
LogPrintf(bs.original);
```